src/jace/translator/jace_subtranslator_interface.py

```diff
@@ -0,0 +1,148 @@
+# JaCe - JAX Just-In-Time compilation using DaCe (Data Centric Parallel Programming)
+#
+# Copyright (c) 2024, ETH Zurich
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from collections.abc import Collection, Sequence
+from typing import TYPE_CHECKING, Any
+
+import dace
+from jax import core as jcore
+
+
+if TYPE_CHECKING:
+    from .jaxpr_translator_driver import JaxprTranslationDriver
+
+
+class JaCeSubTranslatorInterface:
```

Comment:
I changed it, and I agree it makes the code a bit more readable.
However, I do not like the name because it suggests that it is able to already translate everything.